adrv9009 cleanup

buha opened the pull request:
This commit sequence simplifies the headless.c and prepares the talise setup sequence to be multi-instance capable.

dbogdan commented on projects/adrv9009/src/app/headless.c:
```diff
@@ -30,6 +30,9 @@
 #include "app_talise.h"
+#ifndef ALTERA_PLATFORM
+#include "xil_cache.h"
+#endif
```
Isn't this an error (the subject of this commit states that some warnings are fixed)? Anyway, I would split this commit into multiple patches and describe exactly what is wrong.

buha replied:
You're right, I will commit separately

dbogdan commented on projects/adrv9009/src.mk:
```diff
@@ -49,31 +49,31 @@ INCS :=	$(PROJECT)/src/app/app_config.h					\
+	$(DRIVERS)/rf-transceiver/talise/talise_user.h			\
+	$(DRIVERS)/rf-transceiver/talise/talise_version.h			\
 	$(PROJECT)/src/firmware/talise_arm_binary.h			\
 	$(PROJECT)/src/firmware/talise_stream_binary.h			\
```
The API is firmware dependent, so both should be kept together.

buha replied:
Correct, it needs to be moved together

dbogdan commented on drivers/Makefile:
```diff
@@ -8,7 +8,8 @@ INCLUDES = -I../include/ \
-	 -I./axi_core/axi_adc_core
+	 -I./axi_core/axi_adc_core \
+	 -I../projects/adrv9009/src/devices/adi_hal
 
```
These changes should go to previous commit, since they are only fixing an error introduced there.

buha replied:
True, I squashed these commits.

buha commented:
There were several forced pushes that changed the following:
- moved `projects/adrv9009/src/firmware` to `drivers/rf-transceiver/talise/firmware`
- squashed the commit that updates `drivers/Makefile` into a more relevant commit
- moved the xil_cache.h include from the "warning fix" commit to a more relevant commit

dbogdan commented:
One more thing: For consistency with the rest of the files, I would keep the init/remove naming instead of init/deinit.

buha commented:
> One more thing: For consistency with the rest of the files, I would keep the init/remove naming instead of init/deinit.

@dbogdan I intentionally used different scheme because these are not drivers, but application level wrappers for initialization setup sequences. app_talise is not even consistent with the rest - the init function is talise_setup, to keep in line with the linux wrapper naming. If you still consider init/remove would be better, I'll do the refactoring.
